app/prisma/schema.prisma

```diff
@@ -611,18 +613,70 @@ model UsageLog {
 
     inputTokens  Int
     outputTokens Int
+    inputCost    Decimal @default(0)
+    outputCost   Decimal @default(0)
     cost         Float
     type         UsageType @default(EXTERNAL)
     billable     Boolean   @default(true)
 
     fineTuneId String?   @db.Uuid
     fineTune   FineTune? @relation(fields: [fineTuneId], references: [id], onDelete: SetNull)
+    baseModel  String?
+    projectId  String?  @db.Uuid
+    project    Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
 
-    projectId String?  @db.Uuid
-    project   Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    invoiceId String?  @db.Uuid
+    invoice  Invoice? @relation(fields: [invoiceId], references: [id], onDelete: SetNull)
 
     createdAt DateTime @default(now())
 
     @@index([fineTuneId, createdAt, type])
     @@index([projectId, createdAt, type])
 }
+
+
+model Invoice {
+    id String @id @default(uuid()) @db.Uuid
+    amount         Decimal @default(0)
+    status         InvoiceStatus @default(PENDING)
+    slug           String
+    billingPeriod  String? // To make invoices more generic. Can we charge clients for something else manually?
+
+    projectId String  @db.Uuid
+    project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+    description Json? // Array of objects: { [key: string]: string | number }
+
+    usageLogs UsageLog[]
+
+    createdAt DateTime @default(now())
+    paidAt DateTime?
+
+    paymentId String?    
+
+    remainingCredits Decimal?
+
+    creditAdjustments CreditAdjustment[]
+
+    @@index([projectId, createdAt])
+}
+
+model CreditAdjustment {
+    id String @id @default(uuid()) @db.Uuid
+    amount         Decimal //Decimal?
+    
+    projectId String  @db.Uuid
+    project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+    description String?
+
+    Invoice Invoice? @relation(fields: [invoiceId], references: [id], onDelete: SetNull)
+    invoiceId String?  @db.Uuid
+}
+
+enum InvoiceStatus {
```

Comment:
Nit: define InvoiceStatus just above Invoice. In most cases we define enums in the schema file just before they are used.

Reply:
Noted, will be done.